---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scEC

<!-- badges: start -->
<!-- badges: end -->

Single-cell Entropic Clustering (scEC) provides an entropy based approach to normalisation, feature selection, differential expression analysis and unsupervised clustering of single-cell RNA-sequencing data.

## Installation

You can install the the development version from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("mjcasy/scEC")
```
## Workflow

The basic workflow is demonstrated on the Tian et al 2018, single-cell mixology data set, a mixture of three cancerous cell lines (https://github.com/LuyiTian/sc_mixology).

```{r, include = FALSE}
Path <- "~/OneDrive - University of Southampton/Data/Tian2018/ThreeClasses/"
```

```{r example}
library(scEC)
library(Matrix)

load(paste0(Path, "CountsMatrix"))

CountsMatrix <- CountsMatrix[rowSums(CountsMatrix) >= 100,]

PopHet <- Population(CountsMatrix)
Mean <- log10(rowMeans(CountsMatrix))
N <- ncol(CountsMatrix)

plot(Mean, PopHet, ylim = c(0, log(N)))
```

```{r}
GOI <- FeatureSelection(CountsMatrix)
CountsMatrix <- CountsMatrix[GOI,]
Mean <- Mean[GOI]

Clustering <- Cluster(CountsMatrix, MaxClus = 3)
InterType <- DifferentialExpression(CountsMatrix, Clustering[,3])

plot(Mean, InterType)
```
