---
output: md_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scEC

<!-- badges: start -->
<!-- badges: end -->

Single-cell Entropic Clustering (scEC) provides an entropy based approach to normalisation, feature selection, differential expression analysis and unsupervised clustering of single-cell RNA-sequencing data.

## Installation

You can install the the development version from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("mjcasy/scEC")
```
## Clustering

The basic workflow is demonstrated on the Tian et al 2018, single-cell mixology data set, a mixture of three cancerous cell lines (https://github.com/LuyiTian/sc_mixology).

```{r, include = FALSE}
Path <- "~/OneDrive - University of Southampton/Data/Tian2018/ThreeClasses/"
```

```{r example, dpi=300}
library(scEC)
library(Matrix)

load(paste0(Path, "CountsMatrix"))

CountsMatrix <- CountsMatrix[rowSums(CountsMatrix) >= 100,]

PopHet <- Population(CountsMatrix)
Mean <- log10(rowMeans(CountsMatrix))
N <- ncol(CountsMatrix)

plot(Mean, PopHet, ylim = c(0, log(N)))
```

```{r, dpi=300}
GOI <- FeatureSelection(CountsMatrix)
CountsMatrix <- CountsMatrix[GOI,]
Mean <- Mean[GOI]

Clustering <- scEC::Cluster(CountsMatrix, numClus = 3)
InterCluster <- DifferentialExpression(CountsMatrix, Clustering)

plot(Mean, InterCluster)
```


## Mapping to reference

```{r, include = FALSE}
Path2 <- "~/OneDrive - University of Southampton/Data/Tian2018/"
```

```{r, message = FALSE}
library("org.Hs.eg.db") 

load(paste0(Path, "CountsMatrix"))
MapCountsMatrix <- CountsMatrix
load(paste0(Path, "Identity"))
MapID <- Identity

load(paste0(Path2, "CountsMatrix"))
ReferenceCountsMatrix <- CountsMatrix
load(paste0(Path2, "Identity"))
ReferenceID <- Identity

symbols <- mapIds(org.Hs.eg.db, 
                  keys = rownames(MapCountsMatrix), 
                  keytype="ENSEMBL",
                  column = "SYMBOL")

syms <- symbols[!is.na(symbols)]

MapCountsMatrix <- MapCountsMatrix[names(syms),]
MapCountsMatrix@Dimnames[[1]] <- unname(syms)

RefGenes <- row.names(ReferenceCountsMatrix)
MapGenes <- row.names(MapCountsMatrix)
CommonGenes <- intersect(RefGenes, MapGenes)

ReferenceCountsMatrix <- ReferenceCountsMatrix[CommonGenes,]
MapCountsMatrix <- MapCountsMatrix[CommonGenes,]
```

```{r}
Ident <- scEC::Map(MapCountsMatrix = MapCountsMatrix,
                   ReferenceCountsMatrix = ReferenceCountsMatrix, 
                   ReferenceID = ReferenceID, 
                   nGenes = 1000)

knitr::kable(table(Ident, MapID))
```
